addons/web/controllers/main.py

```diff
@@ -1110,9 +1110,14 @@ def content_image(self, xmlid=None, model='ir.attachment', id=None, field='datas
                 if height > 500:
                     height = 500
             try:
-                content = odoo.tools.image_resize_image(base64_source=content, size=(width or None, height or None),
-                                                    encoding='base64', upper_limit=upper_limit,
-                                                    avoid_if_small=avoid_if_small)
+                if request.context.get('keep_progressive'):
+                    content = odoo.tools.image_resize_image(base64_source=content, size=(width or None, height or None),
+                                                            encoding='base64', upper_limit=upper_limit,
+                                                            avoid_if_small=avoid_if_small, keep_progressive=True)
+                else:
+                    content = odoo.tools.image_resize_image(base64_source=content, size=(width or None, height or None),
+                                                            encoding='base64', upper_limit=upper_limit,
+                                                            avoid_if_small=avoid_if_small)
```

Comment:
It was hard for me to realize what was the change

I think you can use a pythonistic way to define an extra parameter

I mean,

```python
                resize_kwargs = {'keep_progressive': True} if request.context.get('keep_progressive') else {}
                content = odoo.tools.image_resize_image(base64_source=content, size=(width or None, height or None),
                                                            encoding='base64', upper_limit=upper_limit,
                                                            avoid_if_small=avoid_if_small, **resize_kwargs)
```

Reply:
I like this approach, let me try this on a dev instance before changing this (it has already been included in production)